### PR TITLE
fix: [DHIS2-10398] disable restrictToCaptureScope of capture scope is more than 100 (2.36)

### PR DESCRIPTION
**Pull request by ameenhere**
This PR aims to be a workaround (or a temporary fix) for the server crash issue if the capture scope org unit count is high. 
If the currently logged in user has more than 100 org units in his capture scope, the restrictToCaptureScope feature/parameter will be disabled. 

A long term solution discussed involved switching to a dedicated API to remove the dirty checking issue as well, but that will need a frontend rewrite in addition to the backend changes. Hence this temporary fix needs to be in place for 2.35.2 as well as 2.36.

### CODE REVIEW

**Review by stian-sandvold — APPROVED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/organisationunit/OrganisationUnitService.java [hunk 1]
```diff
@@ -502,4 +502,33 @@ void deleteOrganisationUnit( OrganisationUnit organisationUnit )
...
+    /**
+     * Cached version of {@link #getCaptureOrganisationUnitCount()
+     * getCaptureOrganisationUnitCount} method If not in cache it populates it
+     * using the {@link #getCaptureOrganisationUnitCount()
+     * getCaptureOrganisationUnitCount} method
+     *
+     * @return the count of the organisation units in capture scope of current
+     *         user
+     */
+    int getCaptureOrganisationUnitCountCached();
```
Comment (enricocolasante) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/organisationunit/OrganisationUnitService.java:532]: Why do we have 2 separate methods for cached and not cached service call?
In my opinion the cache should be transparent to the client of the service

Comment (ameenhere) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/organisationunit/OrganisationUnitService.java:532]: I tried to follow the same approach as used in other method in the same class here https://github.com/dhis2/dhis2-core/blob/100ef730eb8eb30254935dbf61de9147275a6c5c/dhis-2/dhis-api/src/main/java/org/hisp/dhis/organisationunit/OrganisationUnitService.java#L388

The client could decide if it needs realtime values or okay with "maybe" stale values. but its arguable I agree.

Comment (enricocolasante) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/organisationunit/OrganisationUnitService.java:532]: Ok, if it is something that is done already and this is a temporary fix, I think is not the time to argue about this

**Review by enricocolasante — APPROVED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/organisationunit/OrganisationUnitService.java (hunk 1 above)
Comment (ameenhere) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/organisationunit/OrganisationUnitService.java]: I have changed it slightly to make it transparent. Also included stians comments.

**Pull request merged**